### PR TITLE
A0-3518: Purge chain-extension: backend 1/2

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

Great fun continued.

Since OBCE is no longer maintained, we have to rework the extension from the very ground. To avoid huge diff, I divided the work into couple of separate changes. This (second) PR:
 - brings back simplified backend part (i.e. the actual handler for extensions); simplified, because still without proper checks, weighting and tests
 - brings back the extension to the runtime

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- I have added tests
- I have made neccessary updates to the Infrastructure
- I have made corresponding changes to the existing documentation
- I have created new documentation

### CODE REVIEW

File: baby-liminal-extension/src/lib.rs [hunk 1]
```diff
@@ -11,13 +11,30 @@
...
 pub type VerificationKeyIdentifier = [u8; 8];
+
+// ------ Backend stuff ----------------------------------------------------------------------------
+
+#[cfg(feature = "runtime")]
+mod backend;
+
+#[cfg(feature = "runtime")]
+pub use backend::BabyLiminalChainExtension;
+#[cfg(feature = "runtime")]
```
Comment (Marcin-Radecki) [inline baby-liminal-extension/src/lib.rs:39]: Why this feature flag is called `runtime`? As this is quite general name, and in reality it is supposed to enable some particular feature regarding to Liminal, why not call it that way for example?

File: baby-liminal-extension/Makefile [hunk 2]
```diff
@@ -9,6 +9,10 @@ lint: ## Check linting rules
 	cargo +nightly fmt --all --check || exit 1
 	cargo +nightly fmt --all --check $(AT_TEST_CONTRACT) || exit 1
 
+clippy: ## Run clippy
```
Comment (Marcin-Radecki) [inline baby-liminal-extension/Makefile:12]: Good idea with Makefile!

Please let us know when  added CI extends over 10 minutes, we'll then split it somehow (so invariant is to not extend a job over 10 minutes please)

File: baby-liminal-extension/src/lib.rs (hunk 1 above)
Comment (Marcin-Radecki) [reply baby-liminal-extension/src/lib.rs:39]: Okay now I see, this is specific to this crate. All good then.

File: bin/runtime/Cargo.toml [hunk 3]
```diff
@@ -157,14 +157,15 @@ runtime-benchmarks = [
 
 # Liminal-related features
 liminal = [
-    #"baby-liminal-extension",
```
Comment (Marcin-Radecki) [inline bin/runtime/Cargo.toml:160]: Ideally if we we'd check if those below two places
* https://github.com/Cardinal-Cryptography/aleph-node/blob/main/.github/workflows/_build-production-node-and-runtime.yml#L30
* https://github.com/Cardinal-Cryptography/aleph-node/blob/main/.github/workflows/_build-test-node-and-runtime.yml#L31

That both runtime and `aleph-node` binary does **not** have any liminal-related symbols

Comment (pmikolajczyk41) [reply bin/runtime/Cargo.toml:160]: checks added

File: .github/workflows/_build-production-node-and-runtime.yml [hunk 4]
```diff
@@ -27,7 +27,10 @@ jobs:
           targets: wasm32-unknown-unknown
 
       - name: Build production binary and runtime
-        run: cargo build --profile production -p aleph-node
+        # yamllint disable-line rule:line-length
+        run: |
+          cargo build --profile production -p aleph-node
+          nm --demangle=rust target/release/aleph-node | grep -q liminal && { echo "Liminal symbols are present"; exit 1; }
```
Comment (Marcin-Radecki) [inline .github/workflows/_build-production-node-and-runtime.yml:33]: I think this should be `target/production/aleph-node` instead `target/release/aleph-node`

Comment (pmikolajczyk41) [reply .github/workflows/_build-production-node-and-runtime.yml]: oh, you're right, thank you for spotting; fixed

**Review by kostekIV — APPROVED** [review]
lgtm

**Review by Yithis — APPROVED** [review]

**Pull request merged**